File: extract_endpoints.py
import json
import sys

def extract_endpoints(items, parent_group=''):
    """Recursively extract all endpoints from Postman collection"""
    endpoints = []
    
    for item in items:
        # If item has a request, it's an endpoint
        if 'request' in item:
            method = item['request'].get('method', 'N/A')
            url = item['request']['url'].get('raw', '').replace('{{base_url}}', '')
            
            # Extract query parameters
            params = []
            if 'query' in item['request']['url']:
                for query_param in item['request']['url']['query']:
                    if 'key' in query_param:
                        params.append(query_param['key'])
            
            # Extract path variables
            if 'variable' in item['request']['url']:
                for var in item['request']['url']['variable']:
                    if 'key' in var:
                        params.append(f":{var['key']}")
            
            endpoints.append({
                'group': parent_group,
                'name': item.get('name', 'N/A'),
                'method': method,
                'url': url,
                'parameters': params
            })
        
        # If item has nested items, recurse
        if 'item' in item:
            new_group = f"{parent_group} > {item.get('name', '')}" if parent_group else item.get('name', '')
            endpoints.extend(extract_endpoints(item['item'], new_group))
    
    return endpoints

# Load the collection
with open('FitArt_JPAS_API_Complete.postman_collection.json', 'r', encoding='utf-8') as f:
    collection = json.load(f)

# Extract all endpoints
endpoints = extract_endpoints(collection.get('item', []))

# Organize by group
grouped = {}
for endpoint in endpoints:
    group = endpoint['group']
    if group not in grouped:
        grouped[group] = []
    grouped[group].append(endpoint)

# Print formatted output
for group in sorted(grouped.keys()):
    print(f"\n{'='*80}")
    print(f"GROUP: {group}")
    print(f"{'='*80}")
    
    for i, endpoint in enumerate(grouped[group], 1):
        print(f"\n{i}. {endpoint['name']}")
        print(f"   Method: {endpoint['method']}")
        print(f"   URL: {endpoint['url']}")
        if endpoint['parameters']:
            print(f"   Parameters: {', '.join(endpoint['parameters'])}")
        else:
            print(f"   Parameters: None")

# Also print summary statistics
print(f"\n\n{'='*80}")
print("SUMMARY")
print(f"{'='*80}")
print(f"Total Groups: {len(grouped)}")
print(f"Total Endpoints: {len(endpoints)}")
print(f"\nEndpoints by Group:")
for group in sorted(grouped.keys()):
    print(f"  {group}: {len(grouped[group])} endpoints")

File: extract_endpoints.php
<?php
$json_file = 'FitArt_JPAS_API_Complete.postman_collection.json';
$json_content = file_get_contents($json_file);
$collection = json_decode($json_content, true);

function extractEndpoints($items, $parentGroup = '') {
    $endpoints = [];
    
    foreach ($items as $item) {
        // If item has a request, it's an endpoint
        if (isset($item['request'])) {
            $method = $item['request']['method'] ?? 'N/A';
            $url = str_replace('{{base_url}}', '', $item['request']['url']['raw'] ?? '');
            
            // Extract parameters
            $params = [];
            if (isset($item['request']['url']['query'])) {
                foreach ($item['request']['url']['query'] as $query_param) {
                    if (isset($query_param['key'])) {
                        $params[] = $query_param['key'];
                    }
                }
            }
            
            // Extract path variables
            if (isset($item['request']['url']['variable'])) {
                foreach ($item['request']['url']['variable'] as $var) {
                    if (isset($var['key'])) {
                        $params[] = ':' . $var['key'];
                    }
                }
            }
            
            $endpoints[] = [
                'group' => $parentGroup,
                'name' => $item['name'] ?? 'N/A',
                'method' => $method,
                'url' => $url,
                'parameters' => $params
            ];
        }
        
        // If item has nested items, recurse
        if (isset($item['item'])) {
            $newGroup = $parentGroup ? $parentGroup . ' > ' . ($item['name'] ?? '') : ($item['name'] ?? '');
            $endpoints = array_merge($endpoints, extractEndpoints($item['item'], $newGroup));
        }
    }
    
    return $endpoints;
}

// Extract endpoints
$endpoints = extractEndpoints($collection['item'] ?? []);

// Organize by group
$grouped = [];
foreach ($endpoints as $endpoint) {
    $group = $endpoint['group'];
    if (!isset($grouped[$group])) {
        $grouped[$group] = [];
    }
    $grouped[$group][] = $endpoint;
}

// Print formatted output
ksort($grouped);

foreach ($grouped as $group => $groupEndpoints) {
    echo "\n" . str_repeat("=", 100) . "\n";
    echo "GROUP: " . $group . "\n";
    echo str_repeat("=", 100) . "\n";
    
    foreach ($groupEndpoints as $i => $endpoint) {
        echo "\n" . ($i + 1) . ". " . $endpoint['name'] . "\n";
        echo "   HTTP Method: " . $endpoint['method'] . "\n";
        echo "   URL Path: " . $endpoint['url'] . "\n";
        if (!empty($endpoint['parameters'])) {
            echo "   Parameters: " . implode(", ", $endpoint['parameters']) . "\n";
        } else {
            echo "   Parameters: None\n";
        }
    }
}

// Print summary
echo "\n\n" . str_repeat("=", 100) . "\n";
echo "SUMMARY\n";
echo str_repeat("=", 100) . "\n";
echo "Total Groups: " . count($grouped) . "\n";
echo "Total Endpoints: " . count($endpoints) . "\n";
echo "\nEndpoints by Group:\n";
foreach ($grouped as $group => $groupEndpoints) {
    echo "  " . $group . ": " . count($groupEndpoints) . " endpoints\n";
}
?>
